mixnet/cmd/queuesim.py

```diff
@@ -0,0 +1,151 @@
+import argparse
+import random
+from copy import deepcopy
+
+import pandas as pd
+import usim
+from numpy.core.fromnumeric import trace
+
+from protocol.nomssip import NomssipConfig
+from protocol.temporalmix import TemporalMixConfig, TemporalMixType
+from queuesim.config import Config
+from queuesim.simulation import Simulation
+from sim.config import LatencyConfig, TopologyConfig
+
+default_config = Config(
+    num_nodes=10,
+    nomssip=NomssipConfig(
+        peering_degree=3,
+        transmission_rate_per_sec=10,
+        msg_size=82,
+        temporal_mix=TemporalMixConfig(
+            mix_type=TemporalMixType.NONE,
+            min_queue_size=10,
+            seed_generator=random.Random(0),
+        ),
+    ),
+    topology=TopologyConfig(
+        seed=random.Random(0),
+    ),
+    latency=LatencyConfig(
+        min_latency_sec=0,
+        max_latency_sec=0,
+        seed=random.Random(0),
+    ),
+    num_sent_msgs=1,
+    msg_interval_sec=1,
+    num_senders=1,
+    sender_generator=random.Random(0),
+)
+
+
+def divides(numbers: list[int], divisors: list[float]) -> list[int]:
+    return [int(n / div) for n, div in zip(numbers, divisors)]
+
+
+num_nodes_params = [20, 40, 80]
+peering_degree_params = divides(num_nodes_params, [5, 4, 2])
+min_queue_size_params = divides(num_nodes_params, [2, 1, 0.5])
+transmission_rate_params = divides(num_nodes_params, [2, 1, 0.5])
+num_sent_msgs_params = [8, 16, 32]
+num_senders_params = divides(num_nodes_params, [10, 5, 2])
```

Comment:
Just something to consider for the future. We could make a `session_parameters` class/struct that will contain all parameters. Then we could execute the experiments just by providing them the `session_parameters`.

Another thing would be to move the parameters to a config file and here have only initialization from the config file. But this is not very important to have if we can define multiple `session_parameters` here.

Reply:
Yes. I will improve this data structure later. Currently, I didn't define a yaml file because I wanted to easily compute parameters based on `num_nodes_params`.

Reply:
Thanks, it's something to keep in mind for future improvements. This might become very handy when we start doing next simulation sessions.